cubed/array_api/creation_functions.py

```diff
@@ -62,10 +98,13 @@ def asarray(
     ):  # pragma: no cover
         return asarray(a.data)
     elif not isinstance(getattr(a, "shape", None), Iterable):
-        # ensure blocks are arrays
+        if dtype is None:
+            dtype = _iterable_to_default_dtype(a, device=device)
```

Comment:
Why do we need to find the dtype here at all - rather than let `nxp.asarray` in the following line resolve it?